### PR TITLE
Linear edits4

### PR DESCRIPTION
**Pull request by zackchase**
revisions to the linear regression with gluon chapter

### CODE REVIEW

File: chapter_linear-networks/linear-regression-gluon.md [hunk 1]
```diff
@@ -20,8 +34,24 @@ labels += nd.random.normal(scale=0.01, shape=labels.shape)
 
 ## Reading Data
 
-Gluon provides the `data` module to read data. Since `data` is often used as a variable name, we will replace it with the pseudonym `gdata` (adding the first letter of Gluon) when referring to the imported `data` module. In each iteration, we will randomly read a mini-batch containing 10 data instances.
-
+Rather than rolling our own diterator,
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-gluon.md]: diterator?

Comment (astonzhang) [reply chapter_linear-networks/linear-regression-gluon.md]: Should that be iterator?

File: chapter_linear-networks/linear-regression-gluon.md [hunk 2]
```diff
@@ -1,10 +1,24 @@
 # Concise Implementation of Linear Regression
 
-With the development of deep learning frameworks, it has become increasingly easy to develop deep learning applications. In practice, we can usually implement the same model, but much more concisely than how we introduce it in the previous section. In this section, we will introduce how to use the Gluon interface provided by MXNet.
+The surge of deep learning has inspired the development 
+of a variety of mature software frameworks,
+that automate much of the repetitive work 
+of implementing deep learning models.
+In the previous section we relied only 
+on NDarray for data storage and linear algebra
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-gluon.md]: Let's use `NDArray` throughout the book.

File: chapter_linear-networks/linear-regression-gluon.md [hunk 3]
```diff
@@ -20,8 +34,24 @@ labels += nd.random.normal(scale=0.01, shape=labels.shape)
...
+we can call upon Gluon's `data` module to read data. 
+Since `data` is often used as a variable name, 
+we will replace it with the pseudonym `gdata` 
+(adding the first letter of Gluon),
+too differentiate the imported `data` module
+from a variable we might define. 
+The first step will be to instantiate an `ArrayDataset`,
+which takes in one or more NDArrays as arguments.
+Here, we pass in `features` and `labels` as arguments.
+Next, we will use the ArrayDataset to instantiate a DataLoader,
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-gluon.md]: `ArrayDataset`

`DataLoader `

File: chapter_linear-networks/linear-regression-gluon.md [hunk 4]
```diff
@@ -20,8 +34,24 @@ labels += nd.random.normal(scale=0.01, shape=labels.shape)
...
+which also requires that we specify a `batch_size` 
+and specify a Boolean value `shuffle` indicating whether or not 
+we want the DataLoader to shuffle the data 
+on each epoch (pass through the dataset).
+
+<!-- In each iteration, we will randomly read a mini-batch containing 10 data instances.
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-gluon.md]: Let's remove

**Comment by zackchase** [comment]
yes

**Pull request merged**